src/App.tsx

```diff
@@ -15,35 +16,112 @@
   'Garlic',
 ];
 
+enum SortField {
+  ALPHABETICAL = 'alphabetical',
+  LENGTH = 'length',
+}
+
+interface SortingOptions {
+  sortField: SortField | null;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortingOptions,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.ALPHABETICAL:
+          return good1.localeCompare(good2);
+        case SortField.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortField | null>(null);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setIsReversed(false);
+    setSortField(null);
+  };
+
+  const isResetVisible = sortField !== null || isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortField.ALPHABETICAL,
+          })}
+          onClick={() => setSortField(SortField.ALPHABETICAL)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortField.LENGTH,
+          })}
+          onClick={() => setSortField(SortField.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={handleReverse}
+        >
           Reverse
         </button>
```

Comment:
The active button indication for reversing the order is implemented correctly using the 'is-light' class when isReversed is false. This aligns with the task requirement to indicate the active button.